src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
+import { Todo } from './types/Todo';
+
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const todos: Todo[] = useAppSelector(state => state.todos);
+  const currentTodo: Todo | null = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useAppDispatch();
+  const set = (todosFromServer: Todo[]) => (
+    dispatch(todosActions.set(todosFromServer))
+  );
+
+  const isLoaded = useMemo((() => todos.length > 0), [todos]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const fetchedTodos: Todo[] = await getTodos();
+
+        set(fetchedTodos);
+      } catch (error) {
+        throw new Error();
```

Comment:
Consider adding a more descriptive error message inside the Error constructor, like 'throw new Error('Failed to fetch todos');'